interfaces/builtin/README.md

```diff
@@ -333,3 +344,240 @@ a store assertion):
+  
+      manual-connected-implicit-slot:
+        allow-installation:
+          slot-snap-type:
+            - core                     # implicit slot
+        deny-auto-connection: true     # force manual connect
+  
+      auto-connected-provided-slot:
+        allow-installation:
+          slot-snap-type:
+            - app                      # app provided slot
+        deny-connection: true          # require allow-connection in snap decl
+  
+      manual-connected-provided-slot:
+        allow-installation:
+          slot-snap-type:
+            - app                      # app provided slot
+        deny-connection: true          # require allow-connection in snap decl
+        deny-auto-connection: true     # force manual connect
+
+As some slots can be provided by both the system or an application snap, some interfaces will be in more than one category.
+
+Auto-connection should be allowed in the base-declaration if the access to the
+resources provided by the interface does not have security implications.
+
+App-provided slots use 'deny-connection: true' since slot implementations
+require privileged access to the system and the snap must be trusted. In this
+manner a snap declaration for the slot-providing snap is required to override
+the base declaration to allow connections with the app-provided slot.
+
+Slots dealing with hardware typically will specify 'gadget' and 'core' as
+the slot-snap-type (eg, serial-port). Eg:
+
+    slots:
+      manual-connected-hw-slot:
+        allow-installation:
+          slot-snap-type:
+            - core
+            - gadget
+        deny-auto-connection: true
+
+Denying auto-connection both stops access but also covers the fact that there
+might be multiple slots for an hardware interface for which there is no way to
+choose one.
+
+So called super-privileged plugs should also be disallowed installation on a
+system and a snap declaration is required to override the base declaration to
+allow installation (eg, kernel-module-control). Eg:
+
+    plugs:
+      manual-connected-super-privileged-plug:
+        allow-installation: false
+        deny-auto-connection: true
+    (remember this overrides slot side rules)
+
+This pattern makes sense for interfaces that carry great security risk and allow to take control outside of the sandbox, e.g. installing a kernel module.
+
+So called super-privileged slot implementations should also be disallowed
+installation on a system and a snap declaration is required to override the
+base declaration to allow installation (eg, docker). Eg:
+
+    slots:
+      manual-connected-super-privileged-slot:
+        allow-installation: false
+        deny-connection: true
+        deny-auto-connection: true
+
+This pattern make sense for interfaces where implementing them requires extensive system access, or there is need for review to check for policy or to avoid resource use/naming clashes.
```

Comment:
```suggestion
This pattern make sense for interfaces where implementing them requires extensive system access, or where there is need for review to check for policy or to avoid resource use/naming clashes.
```